### PR TITLE
Feature/create dynamic form templates

### PR DESCRIPTION
**Pull request by akashP2312**

### CODE REVIEW

**Review by cpandya231 — CHANGES_REQUESTED** [review]
Lets keep submit and cancel button in one line. 



image

File: src/app/forms/form-workflow/add-edit-workflow-state/add-edit-workflow-state.component.html [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            <div class="row">
+                <div class="form-check form-switch col-6">
+                    <label for="triggerSMS" class="col-4 col-form-label">Trigger Notification Via SMS</label>
+                    <input class="form-check-input" type="checkbox" id="triggerSMS" formControlName="triggerSMS">
+                </div>
+            </div>
+            <div class="row btn_container">
+                <div>
+                    <button class="btn" (click)="Cancel()">Cancel</button>
+                    <button class="btn save_button" type="submit"> Save State</button>
```
Comment (cpandya231) [inline src/app/forms/form-workflow/add-edit-workflow-state/add-edit-workflow-state.component.html]: Lets use same call to action button style everywhere. We should use $main-teal for buttons and $main-teal-dark on hover.

**Review by cpandya231 — CHANGES_REQUESTED** [review]

**Comment by cpandya231** [comment]
Also, I can not add sub-state under existing state. It gives error. Looks like d3-org-chart is not getting proper data.

**Comment by akashP2312** [comment]
Please check now

**Comment by cpandya231** [comment]
Some more points

-  We should redirect to Form Workflow tab when form template is created
-  We should put validation on Previous state dropdown, I am able to create state without any previous state and that breaks the UI. 
- To solve above, I think we should by default create one parent state with default name.
- When we add child states, save workflow button disappears.

**Comment by akashP2312** [comment]
Done changes

**Review by cpandya231 — APPROVED** [review]

**Pull request merged**